### PR TITLE
docs: add description of project direction, vision and scope

### PR DESCRIPTION
**Pull request by lizrice**
Cilium’s [requirements for commit access](https://github.com/cilium/community/blob/main/GOVERNANCE.md#granting-commit-access) include “Plans for sustained contribution to the project compatible with the project’s direction as viewed by current committers” but until now we had no clear articulation of what that direction is. This commit adds a description of Cilium's direction, vision and scope (agreed by lazy consensus in the committers Slack channel).

### CODE REVIEW

**Review by qmonnet — APPROVED** [review]
Thank you Liz!

**Review by tklauser — APPROVED** [review]
Thanks

File: VISION.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+While Cilium’s architecture is geared to be aligned with modern orchestration systems such as Kubernetes, the project scope includes providing connectivity and runtime security for all sorts of workloads including other container orchestration systems, virtual machines, physical servers, and arbitrary network endpoints. The goal is to abstract and decouple connectivity, security, and observability from the underlying infrastructure to achieve reduced complexity, increased portability, and simplified operations. Cilium’s scope includes open source solutions for the control and data planes for these capabilities.
```
Comment (xmulligan) [inline VISION.md:5]:
```suggestion
While Cilium’s architecture is geared to be aligned with modern orchestration systems such as Kubernetes, the project scope includes providing connectivity and runtime security for all sorts of workloads including other container orchestration systems, virtual machines, physical servers, and arbitrary network endpoints. The goal is to abstract and decouple connectivity, security, and observability from the underlying infrastructure to achieve reduced complexity, increased portability, and simplified operations. Cilium’s scope includes open source solutions for the control and data planes of these capabilities.
```

**Review by xmulligan — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Comment by oblazek** [comment]
looks good :+1:

**Review by joestringer — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]
(note that this needs a rebase)

**Review by gandro — APPROVED** [review]

**Pull request merged**